### DISCUSSION TITLE
[Question]: Google OAuth not work

### DISCUSSION THREAD

**Discussion by woojuini** [danny-avila/LibreChat#1639] [Troubleshooting] [answered]
### What is your question?

google oauth integration is not working properly.

### More Details

I'm using google oauth for several applications.
I also integrated librechat, but I get an Internal Server Error and authentication is not completed.
In docker log, I see the following logs.

LibreChat      | 2024-01-26 02:08:13 error: [googleLogin] Cannot read properties of null (reading 'includes')
LibreChat      | TypeError: Cannot read properties of null (reading 'includes')
LibreChat      |     at handleExistingUser (/app/api/strategies/process.js:22:34)
LibreChat      |     at Strategy.googleLogin [as _verify] (/app/api/strategies/googleStrategy.js:16:13)
LibreChat      |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

The path to the callback is shown below.
librechat.testdomain.com:3080/oauth/google/callback?code=4aT-9epWR0r_06aFJII9HjXscope=email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+openid+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email&authuser=0&hd=testdomain.com&prompt=none

The scope of the oauth consent menu is also set as below.
<img width="498" alt="image" src="https://github.com/danny-avila/LibreChat/assets/42531581/45363d62-17b6-46b7-81d8-b22b27d455b4">

### What is the main subject of your question?

User System/OAuth

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I think you're missing some of these values, based on the error, probably the DOMAIN ones. It's working, I use it every day.

```bash

DOMAIN_CLIENT=http://localhost:3080
DOMAIN_SERVER=http://localhost:3080

GOOGLE_CLIENT_ID=your_client_id
GOOGLE_CLIENT_SECRET=your_client_secret
GOOGLE_CALLBACK_URL=/oauth/google/callback
```

have you set all of these values?

**Reply by woojuini** [reply]
Thanks for the quick advice.
already set what you said.

DOMAIN_CLIENT=http://librechat.mydomain.net:3080/
DOMAIN_SERVER=http://librechat.mydomain.net:3080/

GOOGLE_CLIENT_ID=my_id.apps.googleusercontent.com
GOOGLE_CLIENT_SECRET=MY_SECRET
GOOGLE_CALLBACK_URL=/oauth/google/callback

Is callback_url /oauth/github/callback instead of /oauth/google/callback/?
when i update callback_url as /oauth/github/callback then got error , Guide documentation seems to be incorrect.

Error: Unknown authentication strategy "github"

**Reply by danny-avila** [reply]
no it should be "google" that's a typo on my end

**Reply by fuegovic** [reply]
if you're setting a custom domain in `DOMAIN_CLIENT` and `DOMAIN_SERVER` you should not need to specify the port.

For example, mine is set like this:
```yaml
DOMAIN_CLIENT=https://my-domain.com
DOMAIN_SERVER=https://my-domain.com
```

**Comment by woojuini** [comment, marked as answer]
I have identified the cause.
I checked the user property in /app/api/strategies/googleStrategy.js and it was set to provider: Local.
The problem was that I had signed up the google account as local before integrating google authentication.

Thank you for all your help.
